### ISSUE TITLE
HTTP status codes in JSON response object (API)

### ISSUE THREAD

**Issue by aapis** [tdwtf/WtfWebApp#141]
I'm building a small Ruby gem for the API and I've had to use the following code to determine an error code.

```ruby
def msg_to_code(message)
  case message
  when 'Invalid Author'
  when 'Invalid Series'
  when 'Invalid Id'
  when 'Invalid Article Slug'
  when 'JSON Serialization Error : '
  when 'No articles found for the current date range or Invalid Series'
  when 'Count cannot be greater than 100'
  when 'Service Unavailable'
    500
  else
    200
  end
end

# called like...
raise APIException, "#{raw[:status]}" if msg_to_code(raw[:status]) == 500

# an example of what 'raw' may look like
# raw = { :status => 'Invalid Id' }
```

Ideally I'd like to see HTTP codes included in the JSON response so I can just do something like:

```ruby
raise "Something bad happened" if raw[:status_code] > 400
```

**Comment by pikle6** [comment]
I agree with you the api would not really be REStful if we dont return proper error codes upon error. I got caught up with something and could give more time. Now that i have started working on it again I guess i can fix this.

**Comment by BenLubar** [comment]
Fixed in 06d5a2bdcdb3b459227a203bf7ecddd185b06ffd
